gtk4/src/subclass/scale.rs

```diff
@@ -0,0 +1,61 @@
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::Cast;
+
+use super::range::RangeImpl;
+use crate::{Range, Scale};
+
+pub trait ScaleImpl: ScaleImplExt + RangeImpl {
+    fn get_layout_offsets(&self, scale: &Self::Type) -> Option<(i32, i32)> {
+        self.parent_get_layout_offsets(scale)
+    }
+}
+
+pub trait ScaleImplExt: ObjectSubclass {
+    fn parent_get_layout_offsets(&self, scale: &Self::Type) -> Option<(i32, i32)>;
+}
+
+impl<T: ScaleImpl> ScaleImplExt for T {
+    fn parent_get_layout_offsets(&self, scale: &Self::Type) -> Option<(i32, i32)> {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkScaleClass;
+            if let Some(f) = (*parent_class).get_layout_offsets {
+                let x = std::ptr::null_mut();
+                let y = std::ptr::null_mut();
```

Comment:
That does not work like this. You pass `NULL` to the C function and it can't modify that as you don't pass a pointer to anything it could modify.

```rust
let mut x = 0;
let mut y = 0;
f(..., &mut x, &mut y);
```

And then this should probably also not be nullable